cmake/GzUtils.cmake

```diff
@@ -1944,6 +1944,116 @@ macro(gz_build_tests)
 
 endmacro()
 
+#################################################
+# gz_build_examples(
+#     SOURCE_DIR <source_dir>
+#     BINARY_DIR <binary_dir>
+#     [COMPONENTS <project_component_dependencies>]
+#     [DEPENDENCIES <project_dependencies>]
+#
+# Build examples for a Gazebo project. Arguments are as follows:
+# Requires a CMakeLists.txt file to be in SOURCE_DIR that acts
+# as a top level project.
+#
+# This generates two test targets
+# * EXAMPLES_Configure_TEST - Equivalent of calling "cmake .." on
+#                             the examples directory
+#
+# * EXAMPLES_Build_TEST - Equivalent of calling "make" on the
+#                         examples directory
+#
+# These tests are run during "make test" or can be run specifically
+# via "ctest -R EXAMPLES_ -V"
+#
+# SOURCE_DIR: Required. Path to the examples folder.
+#             For example ${CMAKE_CURRENT_SOURCE_DIR}/examples
+#
+# BINARY_DIR: Required. Path to the output binary folder
+#             For example ${CMAKE_CURRENT_BINARY_DIR}/examples
+#
+# COMPONENTS: Optional. List of components built by this project
+#             that are used in the examples.
+#             For example "cli" for gz-utils
+#
+# DEPENCENCIES: Optional. Gazebo library dependencies for this
+#               project
```

Comment:
It's not clear to me what dependencies need to go here. I see that for `gz-math` you added

```
    DEPENDENCIES 
      gz-cmake${GZ_CMAKE_VER} 
      gz-utils${GZ_UTILS_VER} 
```

Both which aren't explicit dependencies of the examples themselves:

https://github.com/gazebosim/gz-math/blob/29e3f4182037ba8c0d525932bda5c6055fe1bb7c/examples/CMakeLists.txt#L6-L7

Do we need to repeat here all the gz libraries that the parent project depends on, but not that project itself?